core/os/os.cpp

```diff
@@ -306,6 +306,15 @@ String OS::get_system_dir(SystemDir p_dir) const {
 	return ".";
 }
 
+String OS::get_safe_application_name() const {
+	String an = Globals::get_singleton()->get("application/name");
+	Vector<String> invalid_char = String("\\ / : * ? \" < > |").split(" ");
+	for (int i=0;i<invalid_char.size();i++) {
+		an = an.replace(invalid_char[i],"-");
```

Comment:
How about replacing those characters with an empty string? E.g. a game named ``The Test: Enclosure of <C:\Hacks>!?`` would have a safe name like ``The Test Enclosure of CHacks`` instead of ``The Test- Enclosure of -C--Hacks---``.

If we do so though, we should add a check to prevent users from naming their game with *only* non allowed characters, e.g. ``*?\/`` would end up with an empty application name. So maybe not the greatest idea in the end :)

Also maybe the ``an`` could be stripped to avoid have names like ``The Game ?`` end up with a trailing space in their safe application name.